django/forms/boundfield.py

```diff
@@ -287,12 +287,15 @@ def build_widget_attrs(self, attrs, widget=None):
                 attrs["required"] = True
         if self.field.disabled:
             attrs["disabled"] = True
-        # If a custom aria-describedby attribute is given and help_text is
-        # used, the custom aria-described by is preserved so user can set the
-        # desired order.
-        if custom_aria_described_by_id := widget.attrs.get("aria-describedby"):
-            attrs["aria-describedby"] = custom_aria_described_by_id
-        elif self.field.help_text and self.id_for_label:
+        # If a custom aria-describedby attribute is given (either via the attrs
+        # argument or widget.attrs) and help_text is used, the custom
+        # aria-described by is preserved so user can set the desired order.
+        if (
+            not attrs.get("aria-describedby")
+            and not widget.attrs.get("aria-describedby")
+            and self.field.help_text
+            and self.id_for_label
+        ):
```

Comment:
I decided to merge the check into one. This way, it means if `widget.attrs` has `aria-describedby`, we'll be using the value directly instead of supplying it into the `attrs` that we're building here.

I think it is better because we're doing less "magic", i.e. the logic is closer to what it's been before #16920. 

To me it reads better too, "if `attrs` doesn't have `aria-describedby` and neither does `widget.attrs`, and if the field has `help_text`, then we supply it based on `id_for_label`".